cmd/candidsrv: fix http-proxy configuration

mhilton opened the pull request:
Go 1.16 changed how the HTTP_PROXY environment variable is interpreted
such that it is no longer used for HTTPS connections. The handling for
this setting has therefore been changed to modify the DefaultTransport
rather than the environment.

Also update the core snap used in candid to core18 as 16.04 is going
end-of-life.

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/95/

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/96/

albertodonato commented:
:shipit:

jujugui commented:
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

mhilton committed:
cmd/candidsrv: fix http-proxy configuration

Go 1.16 changed how the HTTP_PROXY environment variable is interpreted
such that it is no longer used for HTTPS connections. The handling for
this setting has therefore been changed to modify the DefaultTransport
rather than the environment.

Also update the core snap used in candid to core18 as 16.04 is going
end-of-life.

alesstimec reviewed (approved):
lgtm

mhilton committed:
update healthcheck
